core/src/Kokkos_CheckUsage.hpp

```diff
@@ -0,0 +1,141 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_CHECK_USAGE_HPP
+#define KOKKOS_CHECK_USAGE_HPP
+
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <variant>
+
+#include <Kokkos_Assert.hpp>
+#include <Kokkos_Macros.hpp>
+#include <Kokkos_ExecPolicy.hpp>
+
+using namespace std::literals;
+
+namespace Kokkos {
+
+[[nodiscard]] bool is_initialized() noexcept;
+[[nodiscard]] bool is_finalized() noexcept;
+
+namespace Impl {
+
+using ExecPolicyVariant =
+    std::variant<Kokkos::ParallelForTag, Kokkos::ParallelScanTag,
+                 Kokkos::ParallelReduceTag>;
```

Comment:
These parallel constructs, not execution policies (`RangePolicy, MDRangePolicy, TeamPolicy`).